ext/ftp/ftp.stub.php

```diff
@@ -35,7 +35,7 @@ function ftp_mkdir($ftp, string $directory): string|false {}
 function ftp_rmdir($ftp, string $directory): bool {}
 
 /** @param resource $ftp */
-function ftp_chmod($ftp, int $mode, string $filename): int|false {}
+function ftp_chmod($ftp, int $permissions, string $remote_filename): int|false {}
```

Comment:
I'm groping in the dark abut what to do with these filename params. When there is only of them, we could only use `$filename`, but it seems helpful for example in `ftp_fput()` to indicate that we need a remote filename. And if we do that, it might be useful to do it everywhere. No idea.. so any suggestion is welcome.

Reply:
How about using source `$source` / `$target`, or maybe `$source_name` / `$target_name` in this extension? It is not consistent with other places where we try to standardize on `$filename`, but I think it makes sense to deviate from that here. The FTP context may already be sufficient to know that a file name is expected.

I think `$target` also makes sense for functions that have no `$source` parameter.

Reply:
I don't think the remote_ prefix here is useful, as most everything in the ftp extension works on remote entities. E.g. $directory in ftp_rmdir above is also really a remote_directory ... same for mkdir, chdir, nlist etc. I would only make the distinction for functions that accept both a local and a remote name.

Reply:
> I don't think the remote_ prefix here is useful, as most everything in the ftp extension works on remote entities.

I can identify myself with this sentiment well, and it seems like the best thing to do.